dataloader/datasets.hpp

```diff
@@ -244,6 +289,50 @@ class Datasets
     mnistDetails.dropHeader = true;
     return mnistDetails;
   }
+
+  const static DatasetDetails<DatasetX, DatasetY> VOCDetection()
+  {
+    DatasetDetails<DatasetX, DatasetY> VOCDetectionDetail(
+        "voc-detection",
+        true,
+        "/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar",
+        "./../data/VOCtrainval_11-May-2012.tar",
+        "504b9278",
+        "image-detection");
+
+    VOCDetectionDetail.trainingImagesPath =
+        "./../data/VOCdevkit/VOC2012/JPEGImages/";
+    VOCDetectionDetail.trainingAnnotationPath =
+      "./../data/VOCdevkit/VOC2012/Annotations/";
+    VOCDetectionDetail.serverName = "http://host.robots.ox.ac.uk";
+    VOCDetectionDetail.PreProcess = PreProcessor<DatasetX, DatasetY>::PascalVOC;
+
+    // Set classes for dataset.
+    VOCDetectionDetail.classes = {"background", "aeroplane", "bicycle",
+      "bird", "boat", "bottle", "bus", "car", "cat", "chair", "cow",
+      "diningtable", "dog", "horse", "motorbike", "person", "pottedplant",
+      "sheep", "sofa", "train", "tvmonitor"};
+
+    return VOCDetectionDetail;
+  }
+
+  const static DatasetDetails<DatasetX, DatasetY> CIFAR10()
+  {
+    DatasetDetails<DatasetX, DatasetY> CIFAR10Detail(
+        "cifar10",
+        true,
+        "mlpack-url",
+        "./../data/cifar10.tar.gz",
+        "correct-hash",
+        "image-classification");
+
+    CIFAR10Detail.trainingImagesPath = "./../data/cifar10/train/";
+
+    CIFAR10Detail.serverName = "ADD-mlpack-URL";
```

Comment:
Ok, the left work is to change this serverName to mlpack's one, right?

Do you have a plan to add more in this PR? (if so, I think it's better to open another PR :D)

Reply:
Other than this, This is ready. I will ask rcurtin to upload the datasets and test sets once everything else is ready.

Reply:
Did you have a chance to look at this?

Reply:
I think this is ready then, I'll ping rcurtin on the IRC

Reply:
I'll increase the number in tests a bit so that we can also use them model testing.

Reply:
I have made the change from my side. It will give a 404 right now but once datasets are up, it should work.